tests/unit_tests/dft/include/compute_tester.hpp

```diff
@@ -0,0 +1,120 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#ifndef ONEMKL_COMPUTE_TESTER_HPP
+#define ONEMKL_COMPUTE_TESTER_HPP
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+#include "oneapi/mkl.hpp"
+#include "test_helper.hpp"
+#include "test_common.hpp"
+#include "reference_dft.hpp"
+
+template <oneapi::mkl::dft::precision precision, oneapi::mkl::dft::domain domain>
+struct DFT_Test {
+    using descriptor_t = oneapi::mkl::dft::descriptor<precision, domain>;
+
+    template <typename ElemT>
+    using usm_allocator_t = sycl::usm_allocator<ElemT, sycl::usm::alloc::shared, 64>;
+
+    using PrecisionType =
+        typename std::conditional_t<precision == oneapi::mkl::dft::precision::SINGLE, float,
+                                    double>;
+
+    using InputType = typename std::conditional_t<domain == oneapi::mkl::dft::domain::REAL,
+                                                  PrecisionType, std::complex<PrecisionType>>;
+    using OutputType = std::complex<PrecisionType>;
+
+    enum class TestType { buffer, usm };
+
+    const std::int64_t size;
+    const std::int64_t conjugate_even_size;
+    static constexpr int error_margin = 10;
```

Comment:
Yes.

Reply:
Yeah, I think it would be better to have a heuristic based on the problem size, like in the MKL examples. And I think we should have a comment that it's a heuristic, average-case error bound.

Reply:
I've set the error margin based on the problem size in https://github.com/oneapi-src/oneMKL/pull/261/commits/75c0fc5bc2aae1238e2a6283da9547736fe8c85e